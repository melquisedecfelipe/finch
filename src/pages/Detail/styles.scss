.item {
  display: flex;

  > .item-container {
    padding: 40px 100px;
    width: 100%;
    position: relative;
    margin-left: 100px;

    > .item-header {
      display: flex;
      justify-content: space-between;
      padding-bottom: 30px;
      border-bottom: 1px solid #cccccc;

      & .item-content {
        display: flex;

        > h1 {
          font-size: 20px;
          line-height: 33px;
          color: #282430;
          font-weight: normal;
        }

        > span {
          font-size: 20px;
          line-height: 33px;
          color: #5e3ba5;
          font-weight: bold;
          margin: 0 20px;
        }
      }

      & p {
        font-size: 12px;
        line-height: 23px;
        color: #8e879e;
      }

      > .item-search {
        display: flex;
        align-items: flex-start;

        & .item-return {
          height: 25px;
          display: flex;
          background: none;
          box-shadow: none;
        }
      }
    }

    > .item-detail {
      display: grid;
      grid-template-columns: 350px 1fr;
      grid-gap: 30px;
      margin-top: 40px;

      & .item-img {
        position: relative;
        width: 350px;
        height: 300px;
        background: no-repeat;
        background-size: cover;
        background-position: center center;
        border-radius: 3px 3px 0px 0px;

        > p {
          position: absolute;
          bottom: 0;
          right: 0;
          font-size: 14px;
          line-height: 21px;
          color: #5e3ba5;
          color: #ffffff;
          padding: 5px 20px;
        }

        > .-red {
          background: transparent linear-gradient(125deg, #ef0e5a 0%, #55118f 100%);
        }

        > .-blue {
          background: transparent linear-gradient(125deg, #1db6ce 0%, #182ac9 100%);
        }
      }

      & h1 {
        font-size: 20px;
        line-height: 33px;
        color: #282430;
      }

      & p {
        font-size: 14px;
        line-height: 23px;
        color: #8e879e;
      }
    }

    > .item-datasheet {
      > h1 {
        font-size: 20px;
        line-height: 33px;
        color: #282430;
        padding: 30px 0;
        border-bottom: 1px solid #cccccc;
      }

      > ul {
        padding: 0;
        margin: 0;
        > li {
          font-size: 14px;
          line-height: 33px;
          color: #8e879e;
        }
      }
    }
  }
}

@media screen and (max-width: 768px) {
  .item {
    flex-direction: column;

    > .item-container {
      padding: 0px;
      margin: 0;

      > .item-header {
        padding: 20px;
        border: none;

        & .item-content {
          display: grid;

          > span {
            margin: 0;
          }
        }
      }

      > .item-detail {
        display: flex;
        flex-direction: column;
        margin: 0;

        & .item-img {
          width: 100%;
          border-radius: 0;
          padding: 0;
        }

        > .item-info {
          padding: 20px;
        }
      }

      > .item-datasheet {
        padding: 20px;
      }
    }
  }
}

@media screen and (min-width: 769px) and (max-width: 992px) {
  .item {
    flex-direction: flex;

    > .item-container {
      > .item-header {
        padding: 0;
        border: none;
      }

      > .item-detail {
        display: flex;
        flex-direction: column;
        margin-top: 20px;

        & .item-img {
          width: 100%;
          border-radius: 0;
        }

        > .item-info {
          margin-top: 20px;
        }
      }

      > .item-datasheet {
        padding: 20px;
      }
    }
  }
}
