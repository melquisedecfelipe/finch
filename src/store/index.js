import { createStore } from 'redux';
// eslint-disable-next-line import/no-extraneous-dependencies
import { composeWithDevTools } from 'redux-devtools-extension';

import rootReducer from './ducks';

const store = createStore(rootReducer, composeWithDevTools());

export default store;
