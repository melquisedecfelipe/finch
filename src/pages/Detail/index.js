import React, { useEffect, useState } from 'react';
import { Link } from 'react-router-dom';

import './styles.scss';

import Header from '../../components/Header';

import Return from '../../assets/return@2x.png';

export default function Detail({ match }) {
  const [item, setItem] = useState([]);
  const localStorageItems = JSON.parse(localStorage.getItem('items'));
  const id = parseInt(match.params.id, 10);

  useEffect(() => {
    async function loadItem() {
      localStorageItems.forEach(elem => {
        if (elem.id === id) {
          setItem(elem);
        }
      });
    }
    loadItem();
  }, []);

  return (
    <div className="item">
      <Header />
      <div className="item-container">
        <div className="item-header">
          <div>
            <div className="item-content">
              <h1>{item.nome}</h1>
              <span>
                {parseInt(item.valor, 10).toLocaleString('pt-br', {
                  style: 'currency',
                  currency: 'BRL',
                })}
              </span>
            </div>
            <p>{item.decricaoCurta}</p>
          </div>
          <div className="item-search">
            <Link to="/">
              <img className="item-return" src={Return} alt="Voltar" />
            </Link>
          </div>
        </div>
        <div className="item-detail">
          <div className="item-img" style={{ backgroundImage: `url(${item.imagem})` }}>
            <p
              className={
                item.exclusivo === true ? '-blue' : '' || item.promocao === true ? '-red' : ''
              }
            >
              {item.exclusivo === true
                ? 'Exclusivo'
                : '' || item.promocao === true
                ? 'Promoção'
                : ''}
            </p>
          </div>
          <div className="item-info">
            <h1>Detalhes do Produto</h1>
            <p>{item.descricaoLonga}</p>
          </div>
        </div>
        {/* <div className="item-datasheet">
          <h1>Ficha Técnica</h1>
          <div>
            <ul>
            </ul>
          </div>
        </div> */}
      </div>
    </div>
  );
}
