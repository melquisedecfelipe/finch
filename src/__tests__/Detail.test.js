import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import Detail from '../pages/Detail/index';

import store from '../store';

describe('Detail page testing with Enzyme', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <Detail />
      </Provider>,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
