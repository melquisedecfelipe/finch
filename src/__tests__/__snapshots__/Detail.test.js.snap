// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail page testing with Enzyme renders without crashing 1`] = `ShallowWrapper {}`;
